### DISCUSSION TITLE
(Feature Request) Copy button on code snippets and code notes

### DISCUSSION THREAD

**Discussion by Br0kenSilos** [TriliumNext/Trilium#5930] [General]
I don't know if this is possible or how hard it would be to implement but, it would be really nice if there was a "copy" button on code snippets and code notes. As an example, Lets say I have a code snippet set as Powershell. (Those new snippets with the syntax highlighting are awesome by the way!) In the upper right hand corner there is a little "tag" stating the snippet is powershell. Would it be possible to add a little link to copy all of the code in that snippet to the clipboard? If adding a link would take up too much room, perhaps the clicking the powershell "tag" itself would just default to copying the snippet?

![image](https://github.com/user-attachments/assets/97b212c1-d8a2-4558-80c9-38fa84f8f3a5)

**Comment by Nriver** [comment]
Check out this widget: [Copy Code Block Widget](https://github.com/Nriver/copy-code-block-widget). Simply double-click a code block to copy its contents!

**Reply by Br0kenSilos** [reply]
Yes. This seems to work good. Thank you for creating and sharing it. Any chance of merging it into the base code for Trilium Next?

**Reply by maphew** [reply]
heh! I love _"Please stand up and stretch your body for a while if you see this :)"_

**Reply by eliandoran** [reply]
@Br0kenSilos , I don't see it merged in TriliumNext unless we implement it as a button instead of double-clicking, in order to be more intuitive for new users.

**Reply by Br0kenSilos** [reply]
@eliandoran , Yes I agree. Perhaps "merge" was the wrong term for me to use. @Nriver 's plugin provides the basic  functionality I am looking for. That being said, I would love to see this kind of functionality, with a designated "copy" button added to Trilium Next. I think an actual built-in function, with a "Copy Code" button would be a great addition to the program and it would provide a very clear purpose for the button.

**Reply by steuerlexi** [reply]
I’d love to see this feature come to Trilium Next as well! A dedicated “Copy Code” button would be super helpful and much more intuitive than double-clicking—especially for newer users. Having it built-in would really polish the experience when working with code snippets. Big thumbs up from me! 👍
